internal/olm/client/client.go

```diff
@@ -214,7 +216,53 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		return curPhase == olmapiv1alpha1.CSVPhaseSucceeded, nil
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
+		depCheckErr := c.getDeploymentStatus(ctx, key, csv)
+		if depCheckErr != nil {
+			return fmt.Errorf("failed to run operator: %v %v", err, depCheckErr)
+		}
+	}
+
+	return err
+}
+
+// Get Deployment status
+
+func (c Client) getDeploymentStatus(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
```

Comment:
After looking through the entire method, `getDeploymentStatus` seems weird since we're not returning the status anywhere. Maybe we're `checkDeploymentStatus` or something like that.

Reply:
The main purpose of this func is to throw back the actual error, as such. I am open if we want to rename to something else.

Reply:
@bharathi-tenneti 

IMO, If the purpose of func is not to GET the Deployment Status then this cannot be its name.  According to your comment `The main purpose of this func is to throw back the actual error, as such. I am open if we want to rename to something else.` it should be something such as checkN,